cdk/src/open-data-platform/lambda/write-demographic-data-handler.ts

```diff
@@ -1,158 +1,224 @@
-                })
-            .on('error',
-                (error) => {
-                  reject(error);
-                })
-            .on('end', () => {
-              console.log('Parsed ' + results.length + ' rows.');
-              resolve(results);
-            });
+      const config: ConnectionPoolConfig = {
+        host: host,
+        port: port,
+        user: username,
+        password: password,
+      };
+
+      console.log('Connecting to database...');
+      let connectionsCount = 0;
+
+      let pool = createConnectionPool({
+        ...config,
+        onError: (err: Error) => {
+            console.log(`${new Date().toISOString()} ERROR - ${err.message}`)},
+        onConnectionOpened: () => {
+          console.log(
+              `Opened connection. Active connections = ${++connectionsCount}`,
+          );
+        },
+        onConnectionClosed: () => {
+          console.log(
+              `Closed connection. Active connections = ${--connectionsCount}`,
+          );
+        },
+        onQueryStart: (_query, {text, values}) => {
+          console.log(
+              `${new Date().toISOString()} START QUERY ${text} - ${
+                  JSON.stringify(
+                      values,
+                      )}`,
+          );
+        },
+        onQueryResults: (_query, {text}, results) => {
+          console.log(
+              `${new Date().toISOString()} END QUERY   ${text} - ${
+                  results.length} results`,
+          );
+        },
+        onQueryError: (_query, {text}, err) => {
+          console.log(
+              `${new Date().toISOString()} ERROR QUERY ${text} - ${
+                  err.message}`,
+          );
+        },
       });
+      console.log('Finished connecting to database...');
+      resolve(pool);
+    } catch (error) {
+      reject(error);
     }
+  });
+}
 
-// Executes SQL statement argument against the MainCluster db.
-const executeSqlStatement = async (secretArn: string, resourceArn: string,
-                                   statement: string,
-                                   db: string = 'postgres'): Promise<Object> => {
-  return new Promise(async function (resolve, reject) {
-    const sqlParams = {
-      secretArn: secretArn,
-      resourceArn: resourceArn,
-      sql: statement,
-      database: db,
-      includeResultMetadata: true
-    };
+// Inserts all rows into the demographics table.
+async function insertRows(
+    db: Queryable, rows: DemographicsTableRow[]): Promise<any[]> {
+  const rowOptions:
+      BulkOperationOptions<'census_geo_id'|'total_population'|
+                           'black_percentage'|'white_percentage'> = {
+        database: db,
+        tableName: `demographics`,
+        columnTypes: {
+          census_geo_id: sql`VARCHAR`,
+          total_population: sql`REAL`,
+          black_percentage: sql`REAL`,
+          white_percentage: sql`REAL`,
+        },
+      };
+
+  return Promise.all([bulkInsert({
```

Comment:
I don't quite understand why this makes me wrap it into Promise.all because bulkInsert returns a promise

Reply:
Hmm yeah you should be able to just `return await bulkInsert`. Perhaps it's because the return type of `insertRows` is an array?